src/context/AppContext.tsx

```diff
@@ -47,34 +49,37 @@ export const AppContextProvider = ({ children }: { children: ReactNode }) => {
   const [userLoading, setUserLoading] = useState(false);
   const [walletConnecting, setWalletConnecting] = useState(false);
 
-  const updateUser = useCallback((address?: string) => {
-    if (address) {
-      try {
-        setUserLoading(true);
-        const apolloClient = getContext(ContextModule.ApolloClient);
-        const query = apolloClient.query<
-          GetCurrentUserQuery,
-          GetCurrentUserQueryVariables
-        >({
-          query: GetCurrentUserDocument,
-          variables: { address },
-          fetchPolicy: 'network-only',
-        });
-        query.then(({ data }) => {
+  const updateUser = useCallback(
+    async (address?: string, shouldBackgroundUpdate = false) => {
+      if (address) {
+        try {
+          if (!shouldBackgroundUpdate) {
+            setUserLoading(true);
+          }
+          const apolloClient = getContext(ContextModule.ApolloClient);
+          const { data } = await apolloClient.query<
```

Comment:
Using `async / await` here instead of  `.then` forces React to pause code execution until the query has completed, which stops React from batching the `setUserLoading` calls. This ensures `userLoading` actually gets set to true, which wasn't happening before.